[
    {
        "content": "<p>I'm trying to reproduce <a href=\"https://github.com/rust-lang-ci/rust/runs/5743988544?check_suite_focus=true\">this try run failure</a>, but I'm not getting anywhere. Is there any way to run as close to the try as possible, or at least get some logging out so I can see if what I think is the issue actually is?</p>",
        "id": 277075341,
        "sender_full_name": "Quy Nguyen",
        "timestamp": 1648594773
    },
    {
        "content": "<p>if you find one please let me know lol</p>",
        "id": 277076050,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648595310
    },
    {
        "content": "<p>If you want to recreate the environment exactly, you can run the docker image.  That is, in your source tree, run <code>src/ci/docker/run.sh dist-x86_64-linux</code>.   There is documentation at <a href=\"https://rustc-dev-guide.rust-lang.org/tests/docker.html\">https://rustc-dev-guide.rust-lang.org/tests/docker.html</a> for more information.</p>\n<p>Another approach is to just use the same config options.  Towards the beginning of the build logs you can find the exact <code>config.toml</code> settings.  Look for the line \"configure: processing command line\", and it lists them out after that.</p>",
        "id": 277082036,
        "sender_full_name": "Eric Huss",
        "timestamp": 1648601075
    },
    {
        "content": "<p>(note that the config options are not always sufficient, especially when modifying bootstrap itself: CI runs builds in a read only checkout so any modifications to the source at build time give a hard error)</p>",
        "id": 277089936,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648610011
    },
    {
        "content": "<p>hmm, <code>src/ci/docker/run.sh --dev aarch64-gnu</code> tells me I need to be on a host <code>aarch64</code> platform though :(</p>",
        "id": 277091378,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648611898
    },
    {
        "content": "<p>I hacked it to think I'm on an aarch host and now I get completely unrelated errors :(</p>\n<div class=\"codehilite\" data-code-language=\"Rust\"><pre><span></span><code><span class=\"n\">Attempting</span><span class=\"w\"> </span><span class=\"n\">with</span><span class=\"w\"> </span><span class=\"n\">retry</span>: <span class=\"nc\">make</span><span class=\"w\"> </span><span class=\"n\">prepare</span><span class=\"w\"></span>\n<span class=\"n\">Traceback</span><span class=\"w\"> </span><span class=\"p\">(</span><span class=\"n\">most</span><span class=\"w\"> </span><span class=\"n\">recent</span><span class=\"w\"> </span><span class=\"n\">call</span><span class=\"w\"> </span><span class=\"n\">last</span><span class=\"p\">)</span>:\n  <span class=\"nc\">File</span><span class=\"w\"> </span><span class=\"s\">\"/checkout/src/bootstrap/bootstrap.py\"</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"n\">line</span><span class=\"w\"> </span><span class=\"mi\">1342</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"k\">in</span><span class=\"w\"> </span><span class=\"o\">&lt;</span><span class=\"n\">module</span><span class=\"o\">&gt;</span><span class=\"w\"></span>\n<span class=\"w\">    </span><span class=\"n\">main</span><span class=\"p\">()</span><span class=\"w\"></span>\n<span class=\"w\">  </span><span class=\"n\">File</span><span class=\"w\"> </span><span class=\"s\">\"/checkout/src/bootstrap/bootstrap.py\"</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"n\">line</span><span class=\"w\"> </span><span class=\"mi\">1325</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"k\">in</span><span class=\"w\"> </span><span class=\"n\">main</span><span class=\"w\"></span>\n<span class=\"w\">    </span><span class=\"n\">bootstrap</span><span class=\"p\">(</span><span class=\"n\">help_triggered</span><span class=\"p\">)</span><span class=\"w\"></span>\n<span class=\"w\">  </span><span class=\"n\">File</span><span class=\"w\"> </span><span class=\"s\">\"/checkout/src/bootstrap/bootstrap.py\"</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"n\">line</span><span class=\"w\"> </span><span class=\"mi\">1290</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"k\">in</span><span class=\"w\"> </span><span class=\"n\">bootstrap</span><span class=\"w\"></span>\n<span class=\"w\">    </span><span class=\"n\">lock</span><span class=\"w\"> </span><span class=\"o\">=</span><span class=\"w\"> </span><span class=\"n\">acquire_lock</span><span class=\"p\">(</span><span class=\"n\">build</span><span class=\"p\">.</span><span class=\"n\">build_dir</span><span class=\"p\">)</span><span class=\"w\"></span>\n<span class=\"w\">  </span><span class=\"n\">File</span><span class=\"w\"> </span><span class=\"s\">\"/checkout/src/bootstrap/bootstrap.py\"</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"n\">line</span><span class=\"w\"> </span><span class=\"mi\">27</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"k\">in</span><span class=\"w\"> </span><span class=\"n\">acquire_lock</span><span class=\"w\"></span>\n<span class=\"w\">    </span><span class=\"n\">con</span><span class=\"w\"> </span><span class=\"o\">=</span><span class=\"w\"> </span><span class=\"n\">sqlite3</span><span class=\"p\">.</span><span class=\"n\">Connection</span><span class=\"p\">(</span><span class=\"n\">path</span><span class=\"p\">,</span><span class=\"w\"> </span><span class=\"n\">timeout</span><span class=\"o\">=</span><span class=\"mi\">0</span><span class=\"p\">)</span><span class=\"w\"></span>\n<span class=\"n\">sqlite3</span><span class=\"p\">.</span><span class=\"n\">OperationalError</span>: <span class=\"nc\">unable</span><span class=\"w\"> </span><span class=\"n\">to</span><span class=\"w\"> </span><span class=\"n\">open</span><span class=\"w\"> </span><span class=\"n\">database</span><span class=\"w\"> </span><span class=\"n\">file</span><span class=\"w\"></span>\n</code></pre></div>\n<p>I checked and it's trying to open <code>/checkout/obj/build</code>, and build is writable. Not sure what's going wrong.</p>",
        "id": 277091925,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648612741
    },
    {
        "content": "<p>ah, I needed to run <code>chown user build</code>.</p>",
        "id": 277092049,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648612854
    },
    {
        "content": "<p><span class=\"user-mention silent\" data-user-id=\"232545\">Joshua Nelson</span> <a href=\"#narrow/stream/182449-t-compiler.2Fhelp/topic/How.20to.20reproduce.20try.20run.20error/near/277091378\">said</a>:</p>\n<blockquote>\n<p>hmm, <code>src/ci/docker/run.sh --dev aarch64-gnu</code> tells me I need to be on a host <code>aarch64</code> platform though :(</p>\n</blockquote>\n<p>ah, well, I guess this is why lol</p>\n<div class=\"codehilite\" data-code-language=\"Rust\"><pre><span></span><code><span class=\"n\">OSError</span>: <span class=\"p\">[</span><span class=\"n\">Errno</span><span class=\"w\"> </span><span class=\"mi\">8</span><span class=\"p\">]</span><span class=\"w\"> </span><span class=\"n\">Exec</span><span class=\"w\"> </span><span class=\"n\">format</span><span class=\"w\"> </span><span class=\"n\">error</span>: <span class=\"o\">'/</span><span class=\"n\">checkout</span><span class=\"o\">/</span><span class=\"n\">obj</span><span class=\"o\">/</span><span class=\"n\">build</span><span class=\"o\">/</span><span class=\"n\">aarch64</span><span class=\"o\">-</span><span class=\"n\">unknown</span><span class=\"o\">-</span><span class=\"n\">linux</span><span class=\"o\">-</span><span class=\"n\">gnu</span><span class=\"o\">/</span><span class=\"n\">stage0</span><span class=\"o\">/</span><span class=\"n\">bin</span><span class=\"o\">/</span><span class=\"n\">cargo</span><span class=\"o\">'</span><span class=\"w\"></span>\n</code></pre></div>",
        "id": 277092163,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648613031
    },
    {
        "content": "<p>ok, I am feeling stuck. I tried the docker approach, which failed as you can see above. I tried copying the x.py command I saw, which didn't reproduce the failure. Finally I tried running the <code>aarch64-gnu</code> job in PR CI, which has been stuck on \"Waiting for a runner to pick up this job...\" for nearly 24 hours :( <a href=\"https://github.com/rust-lang/rust/runs/5748319740?check_suite_focus=true\">https://github.com/rust-lang/rust/runs/5748319740?check_suite_focus=true</a></p>\n<p>Does anyone have suggestions? the only way I've seen this fail so far is after it's been approved in a full bors test suite, which seems ... not great, since I need reapproval each time I want to change, and each change <em>has</em> to be ready to land as is without cleanups</p>",
        "id": 277207099,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648676116
    },
    {
        "content": "<p>it's possible I just got the CI change wrong? not sure <a href=\"https://github.com/rust-lang/rust/pull/95450/commits/7cdc60f8492f62734616192f66c7232b7705cf81\">https://github.com/rust-lang/rust/pull/95450/commits/7cdc60f8492f62734616192f66c7232b7705cf81</a></p>",
        "id": 277207150,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648676158
    },
    {
        "content": "<p>I tried running the docker and got a bunch of apt-get errors, but that's possibly because my WSL is cursed. I think try runs are a way to test it in actual CI tho (that's how I've been testing mine), but having to ping someone every change just feels like I'm taking up someone's time.</p>",
        "id": 277215178,
        "sender_full_name": "Quy Nguyen",
        "timestamp": 1648682567
    },
    {
        "content": "<p>try runs do not catch the bug in my PR :(</p>",
        "id": 277215207,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648682619
    },
    {
        "content": "<p>you can modify the ci.yml to also execute the aarch builder in try</p>",
        "id": 277238771,
        "sender_full_name": "The 8472",
        "timestamp": 1648708598
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"330154\">@The 8472</span> I tried that, see above. Does it matter whether the job is a PR check or a try check? I don't understand our ci at all really ...</p>",
        "id": 277273908,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648730461
    },
    {
        "content": "<p>PR and try are separate workflows. The former has 3 test jobs, the latter only does a dist for x86_64 linux</p>",
        "id": 277274344,
        "sender_full_name": "The 8472",
        "timestamp": 1648730683
    },
    {
        "content": "<p>Right. My question is whether <code>aarch64-gnu</code> has a different meaning depending on where in the file it is, since I'm already modifying the CI file.</p>",
        "id": 277275663,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648731344
    },
    {
        "content": "<p>Both derive from base-ci-job so I think they should do the same. Except that try runs on the rust-lang-ci org which has more privileges or something? Not sure how that works.</p>",
        "id": 277277319,
        "sender_full_name": "The 8472",
        "timestamp": 1648732006
    },
    {
        "content": "<p>ah hmm maybe, I imagine that <code>self-hosted</code> might not work outside the org</p>",
        "id": 277277375,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648732034
    },
    {
        "content": "<p>ok, will try moving it to a try job</p>",
        "id": 277277384,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1648732039
    },
    {
        "content": "<p>Oh yeah, maybe the aarch runners aren't available to PR builds.</p>",
        "id": 277277385,
        "sender_full_name": "The 8472",
        "timestamp": 1648732039
    }
]